### PR TITLE
Feature/ensure overlay passes compilation tests

### PR DESCRIPTION
**Pull request by patrick-tolosa**
This PR ensures that Overlay passes our compilation tests.

### CODE REVIEW

File: scripts/createEntrypoints.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ const {
 createEntryPointFromESMPath('actions', 'actions');
 
 // Create an Entry Point from everything under bundles/esm/colors -> dist/colors
-createEntryPointFromESMPath('colors', 'colors');
+createEntryPointFromESMPath('constants', 'constants');
```
Comment (patrick-tolosa) [inline scripts/createEntrypoints.ts:14]: 'colors' was a typo, and the actual value should have always been 'constants'

File: src/actions/layers/Source.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import TransformableImage from "../../transformation/TransformableImage";
```
Comment (maoznir) [inline src/actions/layers/Source.ts:1]: File name should match class name no?

File: scripts/createEntrypoints.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ const {
 createEntryPointFromESMPath('actions', 'actions');
 
 // Create an Entry Point from everything under bundles/esm/colors -> dist/colors
```
Comment (maoznir) [inline scripts/createEntrypoints.ts:13]: You should also update the comment

File: __TESTS__/unit/actions/Overlay.test.ts [hunk 4]
```diff
@@ -14,12 +14,12 @@ describe('Tests for overlay actions', () => {
       })
       .resize(fill(1000, 1000))
       .overlay(
-        overlayLayer(
-          imageSource('sample')
+        imageLayer(
+          image('sample')
```
Comment (maoznir) [inline __TESTS__/unit/actions/Overlay.test.ts:18]: Shouldn't we make it possible to import only source.image instead of whole source ?

Comment (patrick-tolosa) [reply __TESTS__/unit/actions/Overlay.test.ts:18]: Yes, we should, I'm not entirely sure what's the bundle impact on keeping it together.

For now, Source seems to be a dumb wrapper around TransforambleImage.

File: src/actions/layers/Source.ts (hunk 2 above)
Comment (patrick-tolosa) [reply src/actions/layers/Source.ts:1]: Source is not an Entrypoint in itself (Something we might want to discuss)

It's not possible to `import Source from 'base/actions/source'`

instead, it's imported through Layers

It's not possible to `import {Source} from 'base/actions/layers'.

A source on its own is not really an action, so I found it weird having it as an entrypoint.

Comment (maoznir) [reply src/actions/layers/Source.ts:1]: the class ImageSource exists in a file named Source. I would put it in ImageSource.ts file.
unless you want to export ImageSource, TextSource, etc from the Source.ts file

File: __TESTS__/unit/actions/Overlay.test.ts (hunk 4 above)
Comment (maoznir) [reply __TESTS__/unit/actions/Overlay.test.ts:18]: Lets not have a "for now" for overlay's source at this stage, we should finalize it's spec

Comment (maoznir) [reply __TESTS__/unit/actions/Overlay.test.ts:18]: We will have different types of sources - image,text,video,subtitles etc no? lets allow to import only whats needed

File: src/actions/layers/Source.ts (hunk 2 above)
Comment (patrick-tolosa) [reply src/actions/layers/Source.ts:1]: For now, I think  that's how it's going to look, I have a feeling that we'll remove Source from the current place any way due to source being used in other, non `overlay` specific contexts

Such as this:

`.effect(Effect.styleTransfer(Source.image('lighthouse')`

I'm honestly not too sure where to place it right now.

File: __TESTS__/unit/actions/Overlay.test.ts (hunk 4 above)
Comment (patrick-tolosa) [reply __TESTS__/unit/actions/Overlay.test.ts:18]: This is related to the above comment, Obviously that's the goal, but we're missing the right place to put Source right now.

Source is not an action, so I'm wondering where we should place it to make it meaningful export.

File: src/actions/layers/Source.ts (hunk 2 above)
Comment (maoznir) [reply src/actions/layers/Source.ts:1]: We need a Params folder

Comment (maoznir) [reply src/actions/layers/Source.ts:1]: or under Overlay

**Pull request merged**